### PR TITLE
Add --targets to CLI

### PR DESCRIPTION
**Pull request by aardschok**
Solves #334
* Added `--targets` through `publish`
* Added check for `PYBLISH_TARGETS` env variable  
if targets are found register them with the API
* Minor cosmetics

### CODE REVIEW

**Comment by tokejepsen** [comment]
Looks good to me. Maybe we need some tests for this?

**Comment by aardschok** [comment]
I'll setup a test for `targets`

**Comment by tokejepsen** [comment]
Looks good to me. @mottosso ?

File: pyblish/api.py [hunk 1]
```diff
@@ -151,6 +151,11 @@ def __init__():
         if host:
             register_host(host)
 
+    # Register targets for current session
+    for t in os.environ.get("PYBLISH_TARGETS", "").split(os.pathsep):
```
Comment (mottosso) [inline pyblish/api.py]: This is too short of a variable name.

Can we also make it..

```python
if not target:
  continue

register_target(target)
```

To reduce indentation?

Comment (mottosso) [reply pyblish/api.py]: Alternatively..

```python
register_target(target) if target else None
```

**Comment by mottosso** [comment]
Ok, with that minor tweak I'm happy.

File: pyblish/api.py (hunk 1 above)
Comment (aardschok) [reply pyblish/api.py]: If I do that then I want to change the lines (line 150) above it as well to have consistency.
I used this as a template to ensure the logic would be the same:
```python
# Register hosts from environment "PYBLISHHOSTS"
for host in os.environ.get("PYBLISH_HOSTS", "").split(os.pathsep):
    if host:
        register_host(host)
```

These lines would then become:
```python
# Register hosts from environment "PYBLISHHOSTS"
for host in os.environ.get("PYBLISH_HOSTS", "").split(os.pathsep):
    if not host:
        continue
    register_host(host)
```

Agreed?

**Comment by mottosso** [comment]
Thanks!

**Pull request merged**